actionsystemtest/lib/action_system_test/server.rb

```diff
@@ -0,0 +1,34 @@
+require "rack/handler/puma"
+
+module ActionSystemTest
+  class Server # :nodoc:
+    def initialize(port)
+      @port = port
+    end
+
+    def run
+      register
+      setup
+    end
+
+    private
+      def register
+        Capybara.register_server :puma do |app, host|
```

Comment:
I've registered this as `:rails_puma` and fixed the server blocks. I also removed the optional port argument and am allowing puma to pick the port now.